src/viur/core/prototypes/tree.py

```diff
@@ -830,6 +913,89 @@ def onDeleted(self, skelType: SkelType, skel: SkeletonInstance):
         if user := current.user.get():
             logging.info(f"""User: {user["name"]!r} ({user["key"]!r})""")
 
+    def onClone(self, skelType: SkelType, skel: SkeletonInstance, src_skel: SkeletonInstance):
+        """
+            Hook function that is called before cloning an entry.
+
+            It can be overwritten to a module-specific behavior.
+
+            :param skelType: Defines the type of the node that is cloned.
+            :param skel: The new Skeleton that is being created.
+            :param src_skel: The source Skeleton `skel` is cloned from.
+
+            .. seealso:: :func:`clone`, :func:`onCloned`
+        """
+        pass
+
+    @CallDeferred
+    def _clone_recursive(
+        self,
+        skel_type: SkelType,
+        src_key: db.Key,
+        target_key: db.Key,
+        target_repo: db.Key,
+        cursor=None
+    ):
+        """
+        Helper function which is used by default onCloned() to clone a recursive structure.
+        """
+        assert (skel_type := self._checkSkelType(skel_type))
+
+        logging.debug(f"_clone_recursive {skel_type=}, {src_key=}, {target_key=}, {target_repo=}, {cursor=}")
+
+        q = self.cloneSkel(skel_type).all().filter("parententry", src_key).order("sortindex")
+        q.setCursor(cursor)
+
+        count = 0
+        for skel in q.fetch():
+            src_skel = skel
+
+            skel = skel.clone()
+            skel["key"] = None
+            skel["parententry"] = target_key
+            skel["parentrepo"] = target_repo
+
+            self.onClone(skel_type, skel, src_skel=src_skel)
+            logging.debug(f"copying {skel=}")  # this logging _is_ needed, otherwise not all values are being written..
+            assert skel.toDB()
+            self.onCloned(skel_type, skel, src_skel=src_skel)
+            count += 1
+
+        logging.debug(f"_clone_recursive {count=}")
+
+        if cursor := q.getCursor():
+            self._clone_recursive(skel_type, src_key, target_key, target_repo, skel_type, cursor)
+
+    def onCloned(self, skelType: SkelType, skel: SkeletonInstance, src_skel: SkeletonInstance):
+        """
+            Hook function that is called after cloning an entry.
+
+            It can be overwritten to a module-specific behavior.
+
+            By default, when cloning a "node", this function calls :func:`_clone_recursive`
+            which recursively clones the entire structure below this node in the background.
+            If this is not wanted, or wanted by a specific setting, overwrite this function
+            without a super-call.
+
+            :param skelType: Defines the type of the node that is cloned.
+            :param skel: The new Skeleton that was created.
+            :param src_skel: The source Skeleton `skel` was cloned from.
+
+            .. seealso:: :func:`clone`, :func:`onClone`
+        """
```

Comment:
```suggestion
        """
        Hook function that is called after cloning an entry.

        It can be overwritten to a module-specific behavior.

        By default, when cloning a "node", this function calls :func:`_clone_recursive`
        which recursively clones the entire structure below this node in the background.
        If this is not wanted, or wanted by a specific setting, overwrite this function
        without a super-call.

        :param skelType: Defines the type of the node that is cloned.
        :param skel: The new SkeletonInstance that was created.
        :param src_skel: The source SkeletonInstance `skel` was cloned from.

        .. seealso:: :func:`clone`, :func:`onClone`
        """
```